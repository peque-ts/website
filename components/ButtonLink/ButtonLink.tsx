import clsx from 'clsx';
import Link from 'next/link';
import React from 'react';

interface Props {
  to: string;
}

export const ButtonLink: React.FC<Props> = ({ children, to }) => (
  <Link href={to}>
    <a
      className={clsx(
        'inline-block rounded py-3 px-8 text-sm font-semibold',
        'bg-primary-500 shadow-lg shadow-[#E712B433] transition hover:bg-primary-400 hover:shadow-[#E712B455]',
      )}
    >
      {children}
    </a>
  </Link>
);
