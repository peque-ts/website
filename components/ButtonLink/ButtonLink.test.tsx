import { render, screen } from '@testing-library/react';
import React from 'react';

import { ButtonLink } from './ButtonLink';

describe('ButtonLink', () => {
  it('should render', () => {
    const { asFragment } = render(<ButtonLink to="/docs">Check the docs</ButtonLink>);

    expect(screen.getByRole('link')).toHaveTextContent('Check the docs');
    expect(asFragment()).toMatchSnapshot();
  });
});
