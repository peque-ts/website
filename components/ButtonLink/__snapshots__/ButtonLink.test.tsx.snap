// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink should render 1`] = `
<DocumentFragment>
  <a
    class="inline-block rounded py-3 px-8 text-sm font-semibold bg-primary-500 shadow-lg shadow-[#E712B433] transition hover:bg-primary-400 hover:shadow-[#E712B455]"
    href="/docs"
  >
    Check the docs
  </a>
</DocumentFragment>
`;
